### DISCUSSION TITLE
Published an In depth Guide on RNTL (700+ GitHub STAR & ~400 CLAP in Medium)

### DISCUSSION THREAD

**Discussion by anisurrahman072** [callstack/react-native-testing-library#1555] [Show and tell]
### 🚀 I have published an [In Depth Guide on RNTL (React Native Testing Library)](https://github.com/anisurrahman072/React-Native-Advanced-Guide/blob/master/Testing/RNTL-Component-Testing-ultimate-guide.md)

[![image](https://github.com/callstack/react-native-testing-library/assets/40535268/e34f0d34-38c3-4d05-8c45-23023c61b940)](https://github.com/anisurrahman072/React-Native-Advanced-Guide/blob/master/Testing/RNTL-Component-Testing-ultimate-guide.md)

- This Ultimate guide was **published** in my [🔥 **React Native Advanced Guide Book (It secured 600+ GitHub STAR ⭐️)**](https://github.com/anisurrahman072/React-Native-Advanced-Guide)
- The same Guide was also published in [**Medium (BLOG) site (It secured ~400 Clap from devs)**](https://medium.com/@anisurrahmanbup/react-native-testing-ultimate-guide-219a5e7ea5dc)
- This guide covered almost all the parts of RNTL in most easiest way.
- It covered all types of **JEST** config that will be helpful when you will use **RNTL**.
- All I describe in the guide is based on my personal implementation & added tons of screenshots on how each part works in your RN app (iOS & Android)
- My [**React Native Advanced Guide Book**](https://github.com/anisurrahman072/React-Native-Advanced-Guide/tree/master) went viral that includes the Guide on [**RNTL**](https://github.com/anisurrahman072/React-Native-Advanced-Guide/blob/master/Testing/RNTL-Component-Testing-ultimate-guide.md). [Tweet Link on the Book](https://x.com/anis_RNCore/status/1741748480400576921?s=20)

### All the topics that have been covered in this guide

- [React Native Testing Library (RNTL)](https://github.com/anisurrahman072/React-Native-Advanced-Guide/blob/master/Testing/RNTL-Component-Testing-ultimate-guide.md#react-native-testing-library-rntl)

- [Setup JEST config (for RNTL)](https://github.com/anisurrahman072/React-Native-Advanced-Guide/blob/master/Testing/RNTL-Component-Testing-ultimate-guide.md#setup-jest-config-for-rntl)

- [Common Error in JEST config — ES6/Ts/cjs/mts](https://github.com/anisurrahman072/React-Native-Advanced-Guide/blob/master/Testing/RNTL-Component-Testing-ultimate-guide.md#common-error-in-jest-config--es6tscjsmts)

- [Command to run a test file](https://github.com/anisurrahman072/React-Native-Advanced-Guide/blob/master/Testing/RNTL-Component-Testing-ultimate-guide.md#command-to-run-test-file)

- [React Native Testing Library (RNTL) APIs](https://github.com/anisurrahman072/React-Native-Advanced-Guide/blob/master/Testing/RNTL-Component-Testing-ultimate-guide.md#react-native-testing-library-rntl-apis)

- [🪵 RNTL API — render()](https://github.com/anisurrahman072/React-Native-Advanced-Guide/blob/master/Testing/RNTL-Component-Testing-ultimate-guide.md#-rntl-api--render)

- [🐠 TEST Component with Provider — by render() API](https://github.com/anisurrahman072/React-Native-Advanced-Guide/blob/master/Testing/RNTL-Component-Testing-ultimate-guide.md#-test-component-with-provider--by-render-api)

- [🐠 “queries” option — render() API](https://github.com/anisurrahman072/React-Native-Advanced-Guide/blob/master/Testing/RNTL-Component-Testing-ultimate-guide.md#-queries-option--render-api)

- [🐠 “update" option — render() API](https://github.com/anisurrahman072/React-Native-Advanced-Guide/blob/master/Testing/RNTL-Component-Testing-ultimate-guide.md#-update-option--render-api)

- [🐠 “debug" option — render() API](https://github.com/anisurrahman072/React-Native-Advanced-Guide/blob/master/Testing/RNTL-Component-Testing-ultimate-guide.md#-debug-option--render-api)

- [🪵 RNTL API — userEvent()](https://github.com/anisurrahman072/React-Native-Advanced-Guide/blob/master/Testing/RNTL-Component-Testing-ultimate-guide.md#-rntl-api--userevent)

- [🐠 Example of FireEvent() API](https://github.com/anisurrahman072/React-Native-Advanced-Guide/blob/master/Testing/RNTL-Component-Testing-ultimate-guide.md#-example-of-fireevent-api)

- [🪵 RNTL API — waitFor()](https://github.com/anisurrahman072/React-Native-Advanced-Guide/blob/master/Testing/RNTL-Component-Testing-ultimate-guide.md#-rntl-api--waitfor)

- [What is Mocking ?](https://github.com/anisurrahman072/React-Native-Advanced-Guide/blob/master/Testing/RNTL-Component-Testing-ultimate-guide.md#what-is-mocking-)

- [🍁 Difference between jest.fn() & jest.mock()](https://github.com/anisurrahman072/React-Native-Advanced-Guide/blob/master/Testing/RNTL-Component-Testing-ultimate-guide.md#-difference-between-jestfn--jestmock)

- [🍁 How to Mock a Real Function (by jest.fn()) ?](https://github.com/anisurrahman072/React-Native-Advanced-Guide/blob/master/Testing/RNTL-Component-Testing-ultimate-guide.md#-how-to-mock-a-real-function-by-jestfn-)

- [🍁 How to Mock a Native Module (by jest.mock())?](https://github.com/anisurrahman072/React-Native-Advanced-Guide/blob/master/Testing/RNTL-Component-Testing-ultimate-guide.md#-how-to-mock-a-native-module-by-jestmock)

- [🍁 Another example of Native Module Mocking ( jest.mock() )](https://github.com/anisurrahman072/React-Native-Advanced-Guide/blob/master/Testing/RNTL-Component-Testing-ultimate-guide.md#-another-example-of-native-module-mocking--jestmock--)

- [Some Advanced Share about RNTL](https://github.com/anisurrahman072/React-Native-Advanced-Guide/blob/master/Testing/RNTL-Component-Testing-ultimate-guide.md#some-advanced-share-about-rntl)

- [Differentiating between host and composite elements](https://github.com/anisurrahman072/React-Native-Advanced-Guide/blob/master/Testing/RNTL-Component-Testing-ultimate-guide.md#differentiating-between-host-and-composite-elements)

**Comment by thymikee** [comment]
Congrats! What do you expect us to do about it? :)

**Reply by anisurrahman072** [reply]
Hi @thymikee, could you guys please add this guide link to the official RNTL doc?

- I have created a pull request by adding this guide link to the `Getting Started` page of documentation, under the `Example` section.
- Pull Request Link: #1556 

### Change to the Guide
I’m eager to accept any suggestions to modify the guide so that it can help the CallStack community.
